client/client.go

```diff
@@ -166,13 +173,20 @@ func (c *Client) SubscribeTx(ctx context.Context, query TxQuery, results chan<-
 	}
 
 	// start a go routine to parse the incoming data and feed to the results channel
-	go func(in <-chan interface{}) {
-		for elem := range in {
-			// TODO: return actual transaction content as well? not just ID and Result
-			// TODO: safer casting???
-			val := elem.(tmtypes.EventDataTx)
-			res := txResultToCommitResult(val.TxResult)
-			results <- res
+	go func(in <-chan ctypes.ResultEvent) {
+	EventLoop:
+		for {
+			select {
+			case <-ctx.Done():
+				break EventLoop
+			case msg := <-in:
+				// TODO: return actual transaction content as well? not just ID and Result
+				// TODO: safer casting???
+				val := msg.Data.(tmtypes.EventDataTx)
+				res := txResultToCommitResult(val.TxResult)
+				results <- res
```

Comment:
This should do select together with `ctx.Done()`

Reply:
yep, I've found that out too. it might actually fix the coverage as well